backend/src/apiserver/server/experiment_server.go

```diff
@@ -37,9 +37,11 @@ func (s *ExperimentServer) CreateExperiment(ctx context.Context, request *api.Cr
 
 func (s *ExperimentServer) GetExperiment(ctx context.Context, request *api.GetExperimentRequest) (
 	*api.Experiment, error) {
-	err := s.canAccessExperiment(ctx, request.Id)
-	if err != nil {
-		return nil, util.Wrap(err, "Failed to authorize the request.")
+	if !common.IsMultiUserSharedReadMode() {
```

Comment:
nit: shall we move the check inside `canAccessXXXX` methods? Because logically, the new flag also controls access, so it seems to me moving inside `canAccessXXX` method groups related logic better.
What do you think?

Reply:
considering we plan to support write/read separated permission model on top of K8s RBAC, it's reasonable we extend the `canAcessXXX` methods to also accept the verb on the resource?

what do you think? I'm okay with current implementation if you think that would be too much for now.

Reply:
I lean towards waiting until that write/read separation is actually happening. By that time maybe a bigger refactoring is preferable than adding a parameter to `canAccessXXX` 

I can see the benefit of having the read/write parameter for this change -- we can have the two flags sit closer in this PR. However, it also adds some complexity not needed for now. Currently there's only one read case for `canAccessXXX`, all other calls are write cases. `ListXXX` is conceptually also a read but don't call this `canAccessXXX` at all. If we were to open the gate for `ListXXX` in the future, we would again add the flag check outside `canAccessXXX`. 

So considering all these floating pieces, I would suggest delay any refactoring and keep this PR as-is.

Reply:
Thanks for the analysis! I am convinced we should delay this.

/lgtm
/approve